docs/guides/input.md

```diff
@@ -0,0 +1,294 @@
+# Input <Badge text="4.25+" />
+
+[:rocket: Quick Reference](#quick-reference)
+
+When you're building software, it's common to split things up into small pieces. These might be functions in a node process, or components in a React app. These small pieces often need to declare **dependencies** - bits of information/code that they're not responsible for, but that they rely on to serve their purpose.
+
+The best way to declare dependencies to an XState machine is called `input`. You can specify input like so:
+
+```ts
+const machine = createMachine(
+  {
+    entry: (context, event, meta) => {
+      console.log(meta.state.input.lyric); // 'In west philadelphia, born and raised'
+    }
+  },
+  {
+    input: {
+      lyric: 'In west philadelphia, born and raised'
+    }
+  }
+);
+```
+
+Or, by using the `.withConfig` method:
+
+```ts
+const machineWithDifferentInput = machine.withConfig({
+  input: {
+    lyric: 'In the playground, where I spend most of my days'
```

Comment:
Should this use the same lyric example as above, as it’s two different ways to achieve the same goal?

Reply:
No, because the inference is you can override it.